catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/model/Search.java

```diff
@@ -94,6 +108,8 @@
 
     public static final String ATTRIBUTE_INDEXED = "indexed";
 
+    private static final int MAX_EXCEPTION_SCAN_DEPTH = 5;
```

Comment:
We recursively scan the exception causes to attempt to provide a meaningful but safe (sanitized; no implementation details exposed) message to return to the integrator/user. I figured recursing a maximum of 5 levels (or until a `null` cause is encountered) is more than sufficient until we default to the vague `REASON_INTERNAL` message.